share/ruby-install/truffleruby-graalvm/functions.sh

```diff
@@ -20,6 +20,11 @@ if [ "$ruby_version" = "23.0.0" ]; then
 	ruby_archive="${ruby_archive:-graalvm-jdk-17.0.7_${graalvm_platform/darwin/macos}-${graalvm_arch/amd64/x64}_bin.tar.gz}"
 	ruby_mirror="${ruby_mirror:-https://download.oracle.com/graalvm/17/archive}"
 	ruby_url="${ruby_url:-$ruby_mirror/$ruby_archive}"
+elif [ "${ruby_version%%.*}" -ge "23" ]; then # 23.1+
```

Comment:
As a general rule-of-thumb, you should use `[[ ]]` for String comparisons and `(( ))` for arithmetic comparisons.
https://mywiki.wooledge.org/BashFAQ/031

Another benefit of `(( ))` over `[[ ]]`, is you can use regular `>` and `<` comparison operators instead of `-gt` or `-lt`.

Reply:
I added a `truffleruby_major` variable, that feels cleaner and more reliable.

Reply:
You could extract `${ruby_version%%.*}` into another variable to make it more explicit. `(( ))` is used when you want to "interpret" the variable's value as an integer.